FinSearchUnified/Bundle/SearchBundleFindologic/FacetHandler/TextFacetHandler.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic\FacetHandler;
+
+use FinSearchUnified\Bundle\SearchBundleFindologic\PartialFacetHandlerInterface;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Bundle\SearchBundle\Facet\ProductAttributeFacet;
+use Shopware\Bundle\SearchBundle\FacetInterface;
+use Shopware\Bundle\SearchBundle\FacetResult\RadioFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListFacetResult;
+use Shopware\Bundle\SearchBundle\FacetResult\ValueListItem;
+use SimpleXMLElement;
+
+class TextFacetHandler implements PartialFacetHandlerInterface
+{
+    /**
+     * @param FacetInterface $facet
+     * @param Criteria $criteria
+     * @param SimpleXMLElement $filter
+     *
+     * @return RadioFacetResult|ValueListFacetResult|null
+     */
+    public function generatePartialFacet(FacetInterface $facet, Criteria $criteria, SimpleXMLElement $filter)
+    {
+        /** @var ProductAttributeFacet $facet */
+        switch ($facet->getMode()) {
+            case ProductAttributeFacet::MODE_VALUE_LIST_RESULT:
+                $result = $this->createValueListFacetResult($facet, $criteria, $filter);
+                break;
+            case ProductAttributeFacet::MODE_RADIO_LIST_RESULT:
+                $result = $this->createRadioFacetResult($facet, $criteria, $filter);
+                break;
+            default:
+                $result = null;
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param SimpleXMLElement $filter
+     *
+     * @return bool
+     */
+    public function supportsFilter(SimpleXMLElement $filter)
+    {
+        return ((string)$filter->name !== 'cat' && in_array((string)$filter->type, ['select', 'label']));
+    }
+
+    /**
+     * @param FacetInterface $facet
+     * @param Criteria $criteria
+     * @param SimpleXMLElement $filterItems
+     *
+     * @return ValueListItem[]
+     */
+    private function getValueListItems(FacetInterface $facet, Criteria $criteria, SimpleXMLElement $filterItems)
+    {
+        $items = [];
+        $actives = [];
+
+        /** @var ProductAttributeFacet $facet */
+        $condition = $criteria->getCondition($facet->getName());
+        if ($condition !== null) {
+            $actives[] = $condition->getValue();
```

Comment:
This will only work if `ProductAttributeFacet::MODE_RADIO_LIST_RESULT` is used. In any other case `$condition->getValue()` will return an array (even if only a single value is possible). Assign the return value of `$condition->getValue()` directly to `$actives` and perform this check afterwards (outside the if condition):
```
if (!is_array($actives)) {
    $actives = [$actives];
}
```